File: Swift/Example/Objc Example/Objc Example/ViewController.m
//
//  ViewController.m
//  Objc Example
//
//  Created by Ameir Al-Zoubi on 4/26/16.
//  Copyright © 2016 Batoul Apps. All rights reserved.
//

#import "ViewController.h"
#import <CoreLocation/CoreLocation.h>
#import "Adhan/Adhan-Swift.h"

@interface ViewController ()

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    NSCalendar *cal = [[NSCalendar alloc] initWithCalendarIdentifier:NSCalendarIdentifierGregorian];
    NSDateComponents *date = [cal components:NSCalendarUnitDay|NSCalendarUnitMonth|NSCalendarUnitYear fromDate:[NSDate date]];
    
    BACalculationParameters *params = [[BACalculationParameters alloc] initWithMethod:BACalculationMethodMuslimWorldLeague];
    params.madhab = BAMadhabHanafi;
    
    BACoordinates *coordinates = [[BACoordinates alloc] initWithLatitude:35.78056 longitude:-78.6389];
    
    BAPrayerTimes *prayerTimes = [[BAPrayerTimes alloc] initWithCoordinates:coordinates date:date calculationParameters:params];
    
    NSDateFormatter *formatter = [[NSDateFormatter alloc] init];
    formatter.timeZone = [[NSTimeZone alloc] initWithName:@"America/New_York"];
    formatter.timeStyle = NSDateFormatterMediumStyle;
    
    NSLog(@"fajr: %@", [formatter stringFromDate:prayerTimes.fajr]);
    NSLog(@"sunrise: %@", [formatter stringFromDate:prayerTimes.sunrise]);
    NSLog(@"dhuhr: %@", [formatter stringFromDate:prayerTimes.dhuhr]);
    NSLog(@"asr: %@", [formatter stringFromDate:prayerTimes.asr]);
    NSLog(@"maghrib: %@", [formatter stringFromDate:prayerTimes.maghrib]);
    NSLog(@"isha: %@", [formatter stringFromDate:prayerTimes.isha]);
}

@end

File: Swift/Sample/ViewController.swift
//
//  ViewController.swift
//  Sample
//
//  Created by Ameir Al-Zoubi on 2/21/16.
//  Copyright © 2016 Batoul Apps. All rights reserved.
//

import UIKit
import Adhan

class ViewController: UIViewController {

    override func viewDidLoad() {
        super.viewDidLoad()
        
        let date = NSDateComponents()
        date.year = 2016
        date.month = 2
        date.day = 21
        var params = CalculationMethod.MoonsightingCommittee.params
        params.offsets = [0, 0, 0, 0, 0, 0]
        let prayers = PrayerTimes(coordinates: Coordinates(latitude: 51.5155, longitude: -0.0922), date: date, calculationParameters: params)!
        
        NSLog("fajr %@", prayers.fajr)
        NSLog("sunrise %@", prayers.sunrise)
        NSLog("dhuhr %@", prayers.dhuhr)
        NSLog("asr %@", prayers.asr)
        NSLog("maghrib %@", prayers.maghrib)
        NSLog("isha %@", prayers.isha)
    }
}

